### PR TITLE
Fix indentation of condition for disabling repos

### PR DESCRIPTION
**Pull request by darnaut**
## Description
Fix handling of `--enablerepo` and `--disablerepo` options in `yum_package`.

There's an indentation error in `yum_helper.py` which was causing `disable` repo options to be ignored unless it appears last in the `repos` array of the command sent to the Yum helper.  The order of the `repos` field does not guarantee that `disable` is the last element, nor it should since the order matters. Example:

```
[2022-11-03T17:06:48+00:00] TRACE: sending '{"action":"whatavailable","provides":"containerd-1.6.6-1.amzn2","repos":[{"disable":"*"},{"enable":"amzn2extra-docker"}]}' to python helper
```

This was also hiding another problem where the helper failed to re-enable repos when all repos are disabled using a wildcard (e.g. "*") as it attempts to `getRepo('*')` which leads to an error. Since rpmdb is closed before and after every query with an `--enablerepo` or `--disablerepo` option, the code that attempts to restore state is simply removed.

## Related Issue
N/A

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by darnaut** [comment]
I suspect this would still be broken for cases like `--disablerepo==* --enablerepo=foo` and the order changes when being converted to `[{"enable":"foo"}, {"disable":"*"}]` for whatever reason.

File: lib/chef/provider/package/yum/yum_helper.py [hunk 1]
```diff
@@ -123,15 +121,7 @@ def query(base, command):
...
-    # Reset any repos we were passed in enablerepo/disablerepo to the original state in enabled_repos
-    if 'repos' in command:
-        for repo in command['repos']:
-            if 'enable' in repo:
-                if base.repos.getRepo(repo['enable']) not in enabled_repos:
-                    base.repos.disableRepo(repo['enable'])
-        if 'disable' in repo:
-            if base.repos.getRepo(repo['disable']) in enabled_repos:
-                base.repos.enableRepo(repo['disable'])
+    # XXX: Reset any repos we were passed in enablerepo/disablerepo to the original state
```
Comment (tpowell-progress) [inline lib/chef/provider/package/yum/yum_helper.py:124]: I would recommend either deleting the comment or commenting how the exit state of this function resets the enabled/disabled state.

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
Adjust the comment on line 124 vs. leaving an XXX comment.

**Review by jaymzh — APPROVED** [review]
Assuming tests pass.

Also we should backport this to 17 and 16?

**Comment by jaymzh** [comment]
@darnaut - can you fix your tests, please?

**Comment by tpowell-progress** [comment]
INFC-381 created for spell check

**Pull request merged**